tests/helpers.bash

```diff
@@ -568,3 +574,125 @@ function stop_git_daemon() {
     rm -f ${TESTDIR}/git-daemon/pid
   fi
 }
+
+# Bring up a registry server using buildah with vfs and chroot as a cheap
+# substitute for podman, accessible only to user $1 using password $2 on the
+# local system at a dynamically-allocated port.
+# Requires openssl.
+# If a user name and password are supplied as the two parameters, htpasswd
+# should be available so that they can be hashed.
+# Sets REGISTRY_PID, REGISTRY_PORT (to append to "localhost:"), and
+# REGISTRY_DIR (where the CA cert can be found) on success.
+function start_registry() {
+  local REGISTRY_IMAGE=quay.io/libpod/registry:2.8
+  local config='
+version: 0.1
+log:
+  fields:
+    service: registry
+storage:
+  cache:
+    blobdescriptor: inmemory
+  filesystem:
+    rootdirectory: /var/lib/registry
+http:
+  addr: :0
+  headers:
+    X-Content-Type-Options: [nosniff]
+  tls:
+    certificate: /etc/docker/registry/localhost.crt
+    key: /etc/docker/registry/localhost.key
+health:
+  storagedriver:
+    enabled: true
+    interval: 10s
+    threshold: 3
+auth:
+  htpasswd:
+    realm: buildah-realm
+    path: /etc/docker/registry/htpasswd
+'
+  # use the result of "htpasswd -nbB testuser testpassword" if an alternate
+  # username and password are not specified
+  local htpasswd='testuser:$2y$05$OW6Qlf1ygRuqJxt/CYcBq.0MZARYLIvr.mgjksw2m7K4cwWkw3Pda'
+  if test "$1" = testuserfoo && test "$2" = testpassword ; then
+    htpasswd='testuserfoo:$2y$05$Pkly/9sA8iGY..2SaerFH.wr4qGPPaGZ66KMrvVmcCP8gaRPpeoeC'
+  elif test "${1:-testuser}" != testuser && test "${2:-testpassword}" != testpassword ; then
```

Comment:
Isn't this just a complicated way of saying:
```bash
    local testuser="${1:-testuser}"
    local testpass="${2:-testpassword}"
```

Reply:
It's an attempt to sort of future-proof that, but thinking on it more, it's better to just avoid needing `htpasswd`, so I'll drop that case.

Reply:
htpasswd is already a hard requirement for the `buildah-tests` package, in both fedora and rhel, because a registry is already mandatory for running tests (and has been from day one).

Reply:
We check in "authenticate: cert and credentials" that we can successfully authenticate to the registry with a known-good username/password pair, and that we get an error when we intentionally supply values that we know the registry isn't configured to accept.  What types of bugs do you see us catching by randomly selecting unique known-good values for each test?

Reply:
Mostly my paranoia: with static values (credentials, input/output test strings) there's always a chance that you're not really testing what you think you're testing: that there's a leftover process somewhere and you're talking to it instead of the registry/httpserver/whatever that you think you're talking to. With unique values that worry is diminished.

Reply:
Hmm, I still like the idea of reducing the set of requirements that someone needs to have installed in order to run the tests, and it turns out we can generate the hashes for passwords ourselves by wrapping the right function call.  I'll add that and switch the "authenticate: cert and credentials" test to randomize them.